kms/awskms/signer.go

```diff
@@ -0,0 +1,122 @@
+package awskms
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"io"
+
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/crypto/pemutil"
+)
+
+// Signer implements a crypto.Signer using the AWS KMS.
+type Signer struct {
+	service   KeyManagementClient
+	keyID     string
+	publicKey crypto.PublicKey
+}
+
+// NewSigner creates a new signer using a key in the AWS KMS.
+func NewSigner(svc KeyManagementClient, signingKey string) (*Signer, error) {
```

Comment:
We are not "re-parsing" it, this method is used from awskms.CreateSigner, and we just pass the signing key in the request object, so it will be only parsed here.

In the request object, the SigningKey represents different things depending on the kms used, for softkms is a path in the directory, and as it comes from the ca.json, a `string` is a valid type. There's no real benefit of using a custom type for this, that we will have to use it in the cli or anything that uses the authority.Config.

Reply:
We parse it once when getting the public key then again when creating the signer. It's not a big deal was just curious if there was a way we could represent this in a form such that the string didn't need to be walked again.

Reply:
We do not use `GetPublicKey` in our code, but have it in the interface makes total sense as part of the KMS interface.